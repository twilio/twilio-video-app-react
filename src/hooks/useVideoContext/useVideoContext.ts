import { useContext } from 'react';
import { VideoContext } from '../../components/VideoProvider';

export default function useVideoContext() {
  const context = useContext(VideoContext);
  if (!context) {
    throw new Error('useVideoContext must be used within a VideoProvider');
  }
  return context;
}
