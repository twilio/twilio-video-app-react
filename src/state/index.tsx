import React, { createContext, useContext, useReducer, useState } from 'react';
import { RecordingRules, RoomType } from '../types';
import { TwilioError } from 'twilio-video';
import { settingsReducer, initialSettings, Settings, SettingsAction } from './settings/settingsReducer';
import useActiveSinkId from './useActiveSinkId/useActiveSinkId';
import useFirebaseAuth from './useFirebaseAuth/useFirebaseAuth';
import usePasscodeAuth from './usePasscodeAuth/usePasscodeAuth';
import { User } from 'firebase';

export interface StateContextType {
  error: TwilioError | Error | null;
  setError(error: TwilioError | Error | null): void;
  getToken(name: string, room: string, passcode?: string): Promise<{ room_type: RoomType; token: string }>;
  user?: User | null | { displayName: undefined; photoURL: undefined; passcode?: string };
  signIn?(passcode?: string): Promise<void>;
  signOut?(): Promise<void>;
  isAuthReady?: boolean;
  isFetching: boolean;
  activeSinkId: string;
  setActiveSinkId(sinkId: string): void;
  settings: Settings;
  dispatchSetting: React.Dispatch<SettingsAction>;
  roomType?: RoomType;
  updateRecordingRules(room_sid: string, rules: RecordingRules): Promise<object>;
}

export const StateContext = createContext<StateContextType>(null!);

/*
  The 'react-hooks/rules-of-hooks' linting rules prevent React Hooks from being called
  inside of if() statements. This is because hooks must always be called in the same order
  every time a component is rendered. The 'react-hooks/rules-of-hooks' rule is disabled below
  because the "if (process.env.REACT_APP_SET_AUTH === 'firebase')" statements are evaluated
  at build time (not runtime). If the statement evaluates to false, then the code is not
  included in the bundle that is produced (due to tree-shaking). Thus, in this instance, it
  is ok to call hooks inside if() statements.
*/
export default function AppStateProvider(props: React.PropsWithChildren<{}>) {
  const [error, setError] = useState<TwilioError | null>(null);
  const [isFetching, setIsFetching] = useState(false);
  const [activeSinkId, setActiveSinkId] = useActiveSinkId();
  const [settings, dispatchSetting] = useReducer(settingsReducer, initialSettings);
  const [roomType, setRoomType] = useState<RoomType>();

  let contextValue = {
    error,
    setError,
    isFetching,
    activeSinkId,
    setActiveSinkId,
    settings,
    dispatchSetting,
    roomType,
  } as StateContextType;

  if (process.env.REACT_APP_SET_AUTH === 'firebase') {
    contextValue = {
      ...contextValue,
      ...useFirebaseAuth(), // eslint-disable-line react-hooks/rules-of-hooks
    };
  } else if (process.env.REACT_APP_SET_AUTH === 'passcode') {
    contextValue = {
      ...contextValue,
      ...usePasscodeAuth(), // eslint-disable-line react-hooks/rules-of-hooks
    };
  } else {
    contextValue = {
      ...contextValue,
      getToken: async (user_identity, room_name) => {
        const endpoint = process.env.REACT_APP_TOKEN_ENDPOINT || '/token';

        return fetch(endpoint, {
          method: 'POST',
          headers: {
            'content-type': 'application/json',
          },
          body: JSON.stringify({
            user_identity,
            room_name,
            create_conversation: process.env.REACT_APP_DISABLE_TWILIO_CONVERSATIONS !== 'true',
          }),
        }).then(res => res.json());
      },
      updateRecordingRules: async (room_sid, rules) => {
        const endpoint = process.env.REACT_APP_TOKEN_ENDPOINT || '/recordingrules';

        return fetch(endpoint, {
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({ room_sid, rules }),
          method: 'POST',
        })
          .then(async res => {
            const jsonResponse = await res.json();

            if (!res.ok) {
              const recordingError = new Error(
                jsonResponse.error?.message || 'There was an error updating recording rules'
              );
              recordingError.code = jsonResponse.error?.code;
              return Promise.reject(recordingError);
            }

            return jsonResponse;
          })
          .catch(err => setError(err));
      },
    };
  }

  const getToken: StateContextType['getToken'] = (name, room) => {
    setIsFetching(true);
    return contextValue
      .getToken(name, room)
      .then(res => {
        setRoomType(res.room_type);
        setIsFetching(false);
        return res;
      })
      .catch(err => {
        setError(err);
        setIsFetching(false);
        return Promise.reject(err);
      });
  };

  const updateRecordingRules: StateContextType['updateRecordingRules'] = (room_sid, rules) => {
    setIsFetching(true);
    return contextValue
      .updateRecordingRules(room_sid, rules)
      .then(res => {
        setIsFetching(false);
        return res;
      })
      .catch(err => {
        setError(err);
        setIsFetching(false);
        return Promise.reject(err);
      });
  };

  return (
    <StateContext.Provider value={{ ...contextValue, getToken, updateRecordingRules }}>
      {props.children}
    </StateContext.Provider>
  );
}

export function useAppState() {
  const context = useContext(StateContext);
  if (!context) {
    throw new Error('useAppState must be used within the AppStateProvider');
  }
  return context;
}
