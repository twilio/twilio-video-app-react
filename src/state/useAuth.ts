import { useCallback, useEffect, useState } from 'react';
import * as firebase from 'firebase/app';
import 'firebase/auth';

const firebaseConfig = {
  apiKey: 'AIzaSyD3HQ2F0OO1oQgkHxnHoqLEtUhL7laJcAE',
  authDomain: 'video-app-79418.firebaseapp.com',
  databaseURL: 'https://video-app-79418.firebaseio.com',
  storageBucket: 'video-app-79418.appspot.com',
  messagingSenderId: '285008367772',
};

firebase.initializeApp(firebaseConfig);

const provider = new firebase.auth.GoogleAuthProvider();
provider.addScope('https://www.googleapis.com/auth/plus.login');

export default function useAuth(setToken: any) {
  const [user, setUser] = useState<firebase.User | null>(null);
  const [isAuthReady, setIsReady] = useState(false);

  useEffect(() => {
    firebase.auth().onAuthStateChanged(user => {
      setUser(user);
      setIsReady(true);
    });
  }, []);

  const signIn = useCallback(() => {
    return firebase
      .auth()
      .signInWithPopup(provider)
      .then(user => {
        setUser(user.user);
      });
  }, []);

  const signOut = useCallback(() => {
    return firebase
      .auth()
      .signOut()
      .then(() => {
        setUser(null);
        setToken('');
      });
  }, [setToken]);

  return { user, signIn, signOut, isAuthReady };
}
