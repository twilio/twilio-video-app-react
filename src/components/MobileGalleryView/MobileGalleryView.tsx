import { CSSProperties } from 'react';
import clsx from 'clsx';
import { makeStyles, createStyles, Theme, useMediaQuery } from '@material-ui/core';
import Participant from '../Participant/Participant';
import useDominantSpeaker from '../../hooks/useDominantSpeaker/useDominantSpeaker';
import useParticipantContext from '../../hooks/useParticipantsContext/useParticipantsContext';
import useVideoContext from '../../hooks/useVideoContext/useVideoContext';

import { Swiper, SwiperSlide } from 'swiper/react';
import 'swiper/css';
import 'swiper/css/pagination';
import { Pagination } from 'swiper';
import { Participant as IParticipant } from 'twilio-video';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    participantContainer: {
      background: theme.galleryViewBackgroundColor,
      position: 'absolute',
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      height: '100%',
      '& .swiper': {
        height: '100%',
        '--swiper-pagination-bullet-inactive-color': 'white',
      },
      '& .swiper-wrapper': {
        height: '100%',
      },
      '& .swiper-pagination.swiper-pagination-bullets': {
        bottom: '5px',
      },
    },
    isPaginationActive: {
      '& .swiper-slide': {
        // To leave room for the pagination indicators:
        height: 'calc(100% - 21px)',
        paddingBottom: '21px',
      },
    },
    swiperSlide: {
      display: 'flex',
      flexWrap: 'wrap',
      alignSelf: 'center',
      alignContent: 'flex-start',
    },
  })
);

export function MobileGalleryView() {
  const classes = useStyles();
  const isMobileLandscape = useMediaQuery('screen and (orientation: landscape)');
  const { room } = useVideoContext();
  const { mobileGalleryViewParticipants } = useParticipantContext();
  const dominantSpeaker = useDominantSpeaker(true);
  const remoteParticipantCount = mobileGalleryViewParticipants.length;

  const pages: IParticipant[][] = [[]];
  // Add the localParticipant to the front of the array to ensure they are always the first participant:
  pages[0].push(room!.localParticipant);

  for (let i = 0; i < remoteParticipantCount; i++) {
    const pageNumber = Math.floor(i / 6);
    if (!pages[pageNumber]) {
      pages[pageNumber] = [];
    }
    // Each page should have a max of 6 participants:
    if (pages[pageNumber].length < 6) {
      pages[pageNumber].push(mobileGalleryViewParticipants[i]);
    } else {
      pages[pageNumber + 1] = [mobileGalleryViewParticipants[i]];
    }
  }

  const portraitParticipantVideoStyles: CSSProperties = {
    width: remoteParticipantCount < 3 ? '100%' : '50%',
    // The height of each participant's video is determined by the number of participants on the gallery view
    // page. Here the array indices represent a remoteParticipantCount. If the count is 4 or greater,
    // the height will be 33.33%
    height: ['100%', '50%', '33.33%', '50%', '33.33%'][Math.min(remoteParticipantCount, 4)],
    padding: '0.2em',
    boxSizing: 'border-box',
  };

  const landscapeParticipantVideoStyles: CSSProperties = {
    height: remoteParticipantCount <= 3 ? '100%' : '50%',
    width: ['100%', '50%', '33.33%', '25%', '33.33%'][Math.min(remoteParticipantCount, 4)],
    padding: '0.2em 0.1em',
    boxSizing: 'border-box',
  };

  return (
    <div className={clsx(classes.participantContainer, { [classes.isPaginationActive]: remoteParticipantCount > 5 })}>
      <Swiper pagination={true} modules={[Pagination]} className="mySwiper">
        {pages.map((page, i) => (
          <SwiperSlide key={i} className={classes.swiperSlide}>
            {page.map(participant => (
              <div
                style={isMobileLandscape ? landscapeParticipantVideoStyles : portraitParticipantVideoStyles}
                key={participant.sid}
              >
                <Participant
                  participant={participant}
                  isLocalParticipant={room!.localParticipant === participant}
                  isDominantSpeaker={dominantSpeaker === participant}
                />
              </div>
            ))}
          </SwiperSlide>
        ))}
      </Swiper>
    </div>
  );
}
