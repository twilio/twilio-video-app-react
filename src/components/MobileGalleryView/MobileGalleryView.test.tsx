import { MobileGalleryView } from './MobileGalleryView';
import { shallow } from 'enzyme';
import { useAppState } from '../../state';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import useParticipantContext from '../../hooks/useParticipantsContext/useParticipantsContext';
import useVideoContext from '../../hooks/useVideoContext/useVideoContext';

const mockLocalParticipant = { identity: 'test-local-participant', sid: 0 };

jest.mock('swiper/react', () => ({
  Swiper: jest.fn(),
  SwiperSlide: jest.fn(),
}));

jest.mock('swiper', () => ({
  Pagination: jest.fn(),
}));

jest.mock('../../hooks/useParticipantsContext/useParticipantsContext');
jest.mock('../../hooks/useVideoContext/useVideoContext');
jest.mock('../../state');
jest.mock('@material-ui/core/useMediaQuery');

const mockUseVideoContext = useVideoContext as jest.Mock<any>;
const mockUseParticipantContext = useParticipantContext as jest.Mock<any>;
const mockUseAppState = useAppState as jest.Mock<any>;
const mockUseMediaQuery = useMediaQuery as jest.Mock<any>;

mockUseAppState.mockImplementation(() => ({ maxGalleryViewParticipants: 9 }));
mockUseVideoContext.mockImplementation(() => ({
  room: {
    localParticipant: mockLocalParticipant,
  },
}));
mockUseParticipantContext.mockImplementation(() => ({
  mobileGalleryViewParticipants: [],
}));

describe('the MobileGalleryView component', () => {
  describe('in portrait orientation', () => {
    beforeEach(() => {
      mockUseMediaQuery.mockImplementation(() => false);
    });

    it('should render correctly when there is only the localParticipant', () => {
      mockUseParticipantContext.mockImplementation(() => ({
        mobileGalleryViewParticipants: [],
      }));

      const wrapper = shallow(<MobileGalleryView />);
      const containers = wrapper.find('[data-test-id="participantContainer"]');
      expect(containers.length).toBe(1);
      expect(containers.prop('style')).toMatchObject({
        height: '100%',
        width: '100%',
      });
    });

    it('should render correctly when there is one participant plus the localParticipant', () => {
      mockUseParticipantContext.mockImplementation(() => ({
        mobileGalleryViewParticipants: [{ identity: 'test-participant-1', sid: 1 }],
      }));

      const wrapper = shallow(<MobileGalleryView />);
      const containers = wrapper.find('[data-test-id="participantContainer"]');
      expect(containers.length).toBe(2);
      expect(containers.at(0).prop('style')).toMatchObject({
        height: '50%',
        width: '100%',
      });
    });

    it('should render correctly when there are two participants plus the localParticipant', () => {
      mockUseParticipantContext.mockImplementation(() => ({
        mobileGalleryViewParticipants: [
          { identity: 'test-participant-1', sid: 1 },
          { identity: 'test-participant-2', sid: 2 },
        ],
      }));

      const wrapper = shallow(<MobileGalleryView />);
      const containers = wrapper.find('[data-test-id="participantContainer"]');
      expect(containers.length).toBe(3);
      expect(containers.at(0).prop('style')).toMatchObject({
        height: '33.33%',
        width: '100%',
      });
    });

    it('should render correctly when there are 3 participants plus the localParticipant', () => {
      mockUseParticipantContext.mockImplementation(() => ({
        mobileGalleryViewParticipants: [
          { identity: 'test-participant-1', sid: 1 },
          { identity: 'test-participant-2', sid: 2 },
          { identity: 'test-participant-3', sid: 3 },
        ],
      }));

      const wrapper = shallow(<MobileGalleryView />);
      const containers = wrapper.find('[data-test-id="participantContainer"]');
      expect(containers.length).toBe(4);
      expect(containers.at(0).prop('style')).toMatchObject({
        height: '50%',
        width: '50%',
      });
    });

    it('should render correctly when there are 3 or more participants plus the localParticipant', () => {
      mockUseParticipantContext.mockImplementation(() => ({
        mobileGalleryViewParticipants: [
          { identity: 'test-participant-1', sid: 1 },
          { identity: 'test-participant-2', sid: 2 },
          { identity: 'test-participant-3', sid: 3 },
          { identity: 'test-participant-4', sid: 4 },
        ],
      }));

      const wrapper = shallow(<MobileGalleryView />);
      const containers = wrapper.find('[data-test-id="participantContainer"]');
      expect(containers.length).toBe(5);
      expect(containers.at(0).prop('style')).toMatchObject({
        height: '33.33%',
        width: '50%',
      });
    });
  });

  describe('in landscape orientation', () => {
    beforeEach(() => {
      mockUseMediaQuery.mockImplementation(() => true);
    });

    it('should render correctly when there is only the localParticipant', () => {
      mockUseParticipantContext.mockImplementation(() => ({
        mobileGalleryViewParticipants: [],
      }));

      const wrapper = shallow(<MobileGalleryView />);
      const containers = wrapper.find('[data-test-id="participantContainer"]');
      expect(containers.length).toBe(1);
      expect(containers.at(0).prop('style')).toMatchObject({
        height: '100%',
        width: '100%',
      });
    });

    it('should render correctly when there is one participant plus the localParticipant', () => {
      mockUseParticipantContext.mockImplementation(() => ({
        mobileGalleryViewParticipants: [{ identity: 'test-participant-1', sid: 1 }],
      }));

      const wrapper = shallow(<MobileGalleryView />);
      const containers = wrapper.find('[data-test-id="participantContainer"]');
      expect(containers.length).toBe(2);
      expect(containers.at(0).prop('style')).toMatchObject({
        height: '100%',
        width: '50%',
      });
    });

    it('should render correctly when there are two participants plus the localParticipant', () => {
      mockUseParticipantContext.mockImplementation(() => ({
        mobileGalleryViewParticipants: [
          { identity: 'test-participant-1', sid: 1 },
          { identity: 'test-participant-2', sid: 2 },
        ],
      }));

      const wrapper = shallow(<MobileGalleryView />);
      const containers = wrapper.find('[data-test-id="participantContainer"]');
      expect(containers.length).toBe(3);
      expect(containers.at(0).prop('style')).toMatchObject({
        height: '100%',
        width: '33.33%',
      });
    });

    it('should render correctly when there are 3 participants plus the localParticipant', () => {
      mockUseParticipantContext.mockImplementation(() => ({
        mobileGalleryViewParticipants: [
          { identity: 'test-participant-1', sid: 1 },
          { identity: 'test-participant-2', sid: 2 },
          { identity: 'test-participant-3', sid: 3 },
        ],
      }));

      const wrapper = shallow(<MobileGalleryView />);
      const containers = wrapper.find('[data-test-id="participantContainer"]');
      expect(containers.length).toBe(4);
      expect(containers.at(0).prop('style')).toMatchObject({
        height: '50%',
        width: '50%',
      });
    });

    it('should render correctly when there are 3 or more participants plus the localParticipant', () => {
      mockUseParticipantContext.mockImplementation(() => ({
        mobileGalleryViewParticipants: [
          { identity: 'test-participant-1', sid: 1 },
          { identity: 'test-participant-2', sid: 2 },
          { identity: 'test-participant-3', sid: 3 },
          { identity: 'test-participant-4', sid: 4 },
        ],
      }));

      const wrapper = shallow(<MobileGalleryView />);
      const containers = wrapper.find('[data-test-id="participantContainer"]');
      expect(containers.length).toBe(5);
      expect(containers.at(0).prop('style')).toMatchObject({
        height: '50%',
        width: '33.33%',
      });
    });
  });
});
