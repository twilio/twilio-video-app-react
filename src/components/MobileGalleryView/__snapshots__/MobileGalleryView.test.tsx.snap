// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the MobileGalleryView component should render correctly when there are 3 or more participants plus the localParticipant 1`] = `
<div
  className="makeStyles-participantContainer-1"
>
  <mockConstructor
    className="mySwiper"
    modules={
      Array [
        [MockFunction],
      ]
    }
    pagination={true}
  >
    <mockConstructor
      className="makeStyles-swiperSlide-3"
      key="0"
    >
      <div
        key="0"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "33.33%",
            "padding": "0.2em",
            "width": "50%",
          }
        }
      >
        <Memo(Participant)
          isDominantSpeaker={false}
          isLocalParticipant={true}
          participant={
            Object {
              "identity": "test-local-participant",
              "sid": 0,
            }
          }
        />
      </div>
      <div
        key="1"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "33.33%",
            "padding": "0.2em",
            "width": "50%",
          }
        }
      >
        <Memo(Participant)
          isDominantSpeaker={false}
          isLocalParticipant={false}
          participant={
            Object {
              "identity": "test-participant-1",
              "sid": 1,
            }
          }
        />
      </div>
      <div
        key="2"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "33.33%",
            "padding": "0.2em",
            "width": "50%",
          }
        }
      >
        <Memo(Participant)
          isDominantSpeaker={false}
          isLocalParticipant={false}
          participant={
            Object {
              "identity": "test-participant-2",
              "sid": 2,
            }
          }
        />
      </div>
      <div
        key="3"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "33.33%",
            "padding": "0.2em",
            "width": "50%",
          }
        }
      >
        <Memo(Participant)
          isDominantSpeaker={false}
          isLocalParticipant={false}
          participant={
            Object {
              "identity": "test-participant-3",
              "sid": 3,
            }
          }
        />
      </div>
      <div
        key="4"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "33.33%",
            "padding": "0.2em",
            "width": "50%",
          }
        }
      >
        <Memo(Participant)
          isDominantSpeaker={false}
          isLocalParticipant={false}
          participant={
            Object {
              "identity": "test-participant-4",
              "sid": 4,
            }
          }
        />
      </div>
    </mockConstructor>
  </mockConstructor>
</div>
`;

exports[`the MobileGalleryView component should render correctly when there are two participants plus the localParticipant 1`] = `
<div
  className="makeStyles-participantContainer-1"
>
  <mockConstructor
    className="mySwiper"
    modules={
      Array [
        [MockFunction],
      ]
    }
    pagination={true}
  >
    <mockConstructor
      className="makeStyles-swiperSlide-3"
      key="0"
    >
      <div
        key="0"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "33.33%",
            "padding": "0.2em",
            "width": "100%",
          }
        }
      >
        <Memo(Participant)
          isDominantSpeaker={false}
          isLocalParticipant={true}
          participant={
            Object {
              "identity": "test-local-participant",
              "sid": 0,
            }
          }
        />
      </div>
      <div
        key="1"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "33.33%",
            "padding": "0.2em",
            "width": "100%",
          }
        }
      >
        <Memo(Participant)
          isDominantSpeaker={false}
          isLocalParticipant={false}
          participant={
            Object {
              "identity": "test-participant-1",
              "sid": 1,
            }
          }
        />
      </div>
      <div
        key="2"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "33.33%",
            "padding": "0.2em",
            "width": "100%",
          }
        }
      >
        <Memo(Participant)
          isDominantSpeaker={false}
          isLocalParticipant={false}
          participant={
            Object {
              "identity": "test-participant-2",
              "sid": 2,
            }
          }
        />
      </div>
    </mockConstructor>
  </mockConstructor>
</div>
`;

exports[`the MobileGalleryView component should render correctly when there is one participant plus the localParticipant 1`] = `
<div
  className="makeStyles-participantContainer-1"
>
  <mockConstructor
    className="mySwiper"
    modules={
      Array [
        [MockFunction],
      ]
    }
    pagination={true}
  >
    <mockConstructor
      className="makeStyles-swiperSlide-3"
      key="0"
    >
      <div
        key="0"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "50%",
            "padding": "0.2em",
            "width": "100%",
          }
        }
      >
        <Memo(Participant)
          isDominantSpeaker={false}
          isLocalParticipant={true}
          participant={
            Object {
              "identity": "test-local-participant",
              "sid": 0,
            }
          }
        />
      </div>
      <div
        key="1"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "50%",
            "padding": "0.2em",
            "width": "100%",
          }
        }
      >
        <Memo(Participant)
          isDominantSpeaker={false}
          isLocalParticipant={false}
          participant={
            Object {
              "identity": "test-participant-1",
              "sid": 1,
            }
          }
        />
      </div>
    </mockConstructor>
  </mockConstructor>
</div>
`;

exports[`the MobileGalleryView component should render correctly when there is only the localParticipant 1`] = `
<div
  className="makeStyles-participantContainer-1"
>
  <mockConstructor
    className="mySwiper"
    modules={
      Array [
        [MockFunction],
      ]
    }
    pagination={true}
  >
    <mockConstructor
      className="makeStyles-swiperSlide-3"
      key="0"
    >
      <div
        key="0"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "100%",
            "padding": "0.2em",
            "width": "100%",
          }
        }
      >
        <Memo(Participant)
          isDominantSpeaker={false}
          isLocalParticipant={true}
          participant={
            Object {
              "identity": "test-local-participant",
              "sid": 0,
            }
          }
        />
      </div>
    </mockConstructor>
  </mockConstructor>
</div>
`;
